### PR TITLE
x-pack/filebeat/input/entityanalytics/provider/{azuread,okta}: add request tracing support

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

See title.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## How to test this PR locally

Run tests in the okta/internal/okta and azuread/fetcher/graph packages with the `-request_trace` flag to see tracing behaviour.

## Related issues

-

### CODE REVIEW

File: x-pack/filebeat/input/entityanalytics/provider/azuread/fetcher/graph/graph.go [hunk 1]
```diff
@@ -383,6 +398,41 @@ func New(cfg *config.C, logger *logp.Logger, auth authenticator.Authenticator) (
 	return &f, nil
 }
 
+// requestTrace decorates cli with an httplog.LoggingRoundTripper if cfg.Trace
```
Comment (chrisberkhout) [inline x-pack/filebeat/input/entityanalytics/provider/azuread/fetcher/graph/graph.go:401]:
```suggestion
// requestTrace decorates cli with an httplog.LoggingRoundTripper if cfg.Tracer
```

File: x-pack/filebeat/input/entityanalytics/provider/azuread/fetcher/graph/graph.go [hunk 2]
```diff
@@ -383,6 +398,41 @@ func New(cfg *config.C, logger *logp.Logger, auth authenticator.Authenticator) (
...
+		cfg.Tracer.Close()
+	}()
+	core := ecszap.NewCore(
+		ecszap.NewDefaultEncoderConfig(),
+		w,
+		zap.DebugLevel,
+	)
+	traceLogger := zap.New(core)
+
+	const margin = 1e3 // 1OkB ought to be enough room for all the remainder of the trace details.
```
Comment (chrisberkhout) [inline x-pack/filebeat/input/entityanalytics/provider/azuread/fetcher/graph/graph.go:420]: Isn't 1e3 == 1kB?

**Review by chrisberkhout — APPROVED** [review]
LGTM

Couple of minor comments that apply to both.

File: x-pack/filebeat/input/entityanalytics/provider/azuread/fetcher/graph/graph.go (hunk 2 above)
Comment (efd6) [reply x-pack/filebeat/input/entityanalytics/provider/azuread/fetcher/graph/graph.go:420]: Indeed it is :facepalm:.

**Pull request merged**